### PR TITLE
Fix OpenLDAP 2.6.0 recipe

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/openldap/package.py
Comment (scheibelp) [inline var/spack/repos/builtin/packages/openldap/package.py:36]: I think there should be a `conflicts('~static ~shared`)` statement.

File: var/spack/repos/builtin/packages/py-python-ldap/package.py
Comment (scheibelp) [inline var/spack/repos/builtin/packages/py-python-ldap/package.py:32]: (question) Do you know if an option is exposed to `setup.py`? E.g.

```
var/spack/repos/builtin/packages//py-h5py/package.py:        self.setup_py('configure', '--hdf5={0}'.format(spec['hdf5'].prefix),
var/spack/repos/builtin/packages//py-h5py/package.py-                      '--hdf5-version={0}'.format(spec['hdf5'].version))
```

Or @adamjstewart do you know if there is a preferred way to set something like this?

**Review by scheibelp — COMMENTED** [review]
I have a question and a request. I also noticed you specifically requested a review from someone, so I can wait to merge it until they add a review if preferred.

File: var/spack/repos/builtin/packages/py-python-ldap/package.py
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-python-ldap/package.py:32]: The preferred way is to not pass options to `setup.py`. Once #27798 is merged, we won't even be running `setup.py` anymore. Environment variables or patches are fine.

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-python-ldap/package.py:32]: @scheibelp it is not exposed

**Review by scheibelp — DISMISSED** [review]
This looks good to me. Since you requested a review from `@Sinan81`, let me know if you want to wait for them to review this. But I can merge this whenever.

**Comment by iarspider** [comment]
@scheibelp let's wait until moday, and if `@Sinan81` doesn't request any changes - merge it.

File: var/spack/repos/builtin/packages/openldap/package.py
Comment (scheibelp) [inline var/spack/repos/builtin/packages/openldap/package.py:55]: It looks like this is optional with `--enable-wt` and also it looks like it might be off by default. If it is optional, can you add a variant for it?

Comment (iarspider) [reply var/spack/repos/builtin/packages/openldap/package.py:55]: It should be optional, but on my test machine (Ubuntu 20.04) it tries to find WT anyway:
```
     330    checking for WT... no
  >> 331    configure: error: Package requirements (wiredtiger) were not met:
     332
     333    Package 'wiredtiger', required by 'virtual:world', not found
     334
     335    Consider adjusting the PKG_CONFIG_PATH environment variable if you
     336    installed software in a non-standard prefix.
     337
```

I can disable it explicitly since I don't build openldap server

**Review by scheibelp — APPROVED** [review]

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**
